crates/driver/src/util/percent.rs

```diff
@@ -0,0 +1,26 @@
+use thiserror::Error;
+
+/// A percentage value. The value is guaranteed to be in the range [0,1].
+#[derive(Clone, Copy, Debug)]
+pub struct Percent(f64);
```

Comment:
```suggestion
pub struct Percent(pub f64);
```

saves you the get method. If you want to expose different flavors of getting this value (e.g. as a factor 0.1 or as a value 10), I'd use explicit names instead of `get`.

Reply:
I aimed here to force creating `Percent` using `TryFrom` and therefore guarantee always valid inner data.